### PR TITLE
[PM-2676] Fix web set environment urls after state init

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Fix bug where web was not setting deployment urls and was defaulting to US prod.

## Code changes

- `stateService.init()` was triggering `activeAccount$` observable before `environmentService` was able to set deployment urls. Therefore a control variable was introduced to ignore activeAccount broadcasts until urls are set.

## Screenshots

https://github.com/bitwarden/clients/assets/4648522/41024595-2c40-48b6-ac95-fdb1568da73d

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/web/src/app/core/init.service.ts [hunk 1]
```diff
@@ -38,11 +38,14 @@ export class InitService {
 
   init() {
     return async () => {
+      // Workaround to ignore stateService.activeAccount until process.env.URLS are set
```
Comment (jlf0dev) [inline apps/web/src/app/core/init.service.ts:41]: Can you add a `TODO` here that we should refactor this as soon as possible?

File: libs/common/src/platform/services/environment.service.ts [hunk 2]
```diff
@@ -270,7 +275,7 @@ export class EnvironmentService implements EnvironmentServiceAbstraction {
       case Region.SelfHosted:
         // if user saves with empty fields, default to US
         if (this.isEmpty()) {
-          this.setRegion(Region.US);
+          await this.setRegion(Region.US);
```
Comment (jlf0dev) [inline libs/common/src/platform/services/environment.service.ts:278]: If `setRegion` is only used in the environment service, can we set it to private or protected?

Comment (andrebispo5) [reply libs/common/src/platform/services/environment.service.ts:278]: It is also used in the `environmentSelector.component`

File: apps/web/src/app/core/init.service.ts (hunk 1 above)
Comment (trmartin4) [reply apps/web/src/app/core/init.service.ts:41]: If you do, this should do it: https://bitwarden.atlassian.net/browse/PM-2637.

I've added this issue to the scope here.

Comment (andrebispo5) [reply apps/web/src/app/core/init.service.ts:41]: Added TODO with JIRA ticket reference

**Review by jlf0dev — APPROVED** [review]
Looks good to me! The tests failed for some reason so I started a rerun, just want to make sure we didn't miss anything.

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**